### PR TITLE
Follow-up to #56490: Fix build with `cmake -DENABLE_LIBRARIES=0`

### PR DESCRIPTION
**Pull request by rschu1ze**
Follow-up to #56490

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-ch-test-poll2** [comment]
*This is an automated comment for commit 8a3b8245657f82bca5026d6aae999af6288ca1e2 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/57330/8a3b8245657f82bca5026d6aae999af6288ca1e2/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/7023858480">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/57330/8a3b8245657f82bca5026d6aae999af6288ca1e2/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

**Comment by rschu1ze** [comment]
@jkartseva Like to check + approve? Thanks.

**Review by jkartseva — APPROVED** [review]
Thanks, should we follow-up with `DENABLE_LIBRARIES=0` in test configuration? Tests did not catch it.

**Comment by rschu1ze** [comment]
As far as I know, FastTests run on a build with minimal dependencies (no contribs). So that would theoretically affected iff the new table function / storage were called by the set of stateless tests which run during FastTest (I guess this is not the case).

The remaining tests in CI (stateless, stateful, integration, fuzzer, etc.) run on standard full builds, i.e. with simdjson and rapidjson.

Similar build issues were introduced in the past with other submodules. Often, only the few people who build locally `-DENABLE_LIBRARIES` (e.g. me) run into these issues.

File: src/TableFunctions/TableFunctionFuzzJSON.h [hunk 1]
```diff
@@ -5,6 +5,9 @@
 #include <Storages/StorageFuzzJSON.h>
 #include <TableFunctions/ITableFunction.h>
 
+#include "config.h"
```
Comment (jkartseva) [inline src/TableFunctions/TableFunctionFuzzJSON.h:8]: Is this file auto-generated by CMake?

**Comment by jkartseva** [comment]
> As far as I know, FastTests run on a build with minimal dependencies (no contribs). So that would theoretically affected iff the new table function / storage were called by the set of stateless tests which run during FastTest (I guess this isn't the case).

The newly added tests were successfully run by FastTest:
https://github.com/ClickHouse/ClickHouse/actions/runs/7013191499/job/19081512193#step:8:10770
https://github.com/ClickHouse/ClickHouse/actions/runs/7013191499/job/19081512193#step:8:11556

**Comment by rschu1ze** [comment]
I stand corrected, your two new tests indeed run in FastTest. This is the case because FastTest is based on a build with `-DENABLE_LIBRARIES=0` *yet with* `-DENABLE_SIMDJSON`:

https://github.com/ClickHouse/ClickHouse/blob/master/docker/test/fasttest/run.sh#L120
https://github.com/ClickHouse/ClickHouse/blob/master/docker/test/fasttest/run.sh#L180

The build only broke for people who did a pure `-DENABLE_LIBRARIES=0` build.

File: src/TableFunctions/TableFunctionFuzzJSON.h (hunk 1 above)
Comment (rschu1ze) [reply src/TableFunctions/TableFunctionFuzzJSON.h:8]: Yes, this file is generated by CMake from Common/config.h.in (the location of "config.h" is somewhere below the build directory).

**Pull request merged**